src/core/createInstance.js

```diff
@@ -121,7 +126,19 @@ export default (
     // underlying function call.
     // Also note that executeCommand may or may not return a promise.
     new Promise(_resolve => {
-      _resolve(executeCommand(commandName, options));
+      logger.log(`Executing ${commandName} command.`, "Options:", options);
+      _resolve(
+        // Note that promiseAllObject also clones the options object.
+        // This is an important aspect because it serves as taking a snapshot
+        // of options content (particularly important with xdm and data
+        // options where the original objects are more likely to be modified
+        // by the customer over time).
+        // TODO: Add a test for createInstance that ensures options are
+        // being cloned.
+        promiseAllObject(options).then(resolvedOptions => {
```

Comment:
Notice that of `options` has a promise somewhere in it, we're not logging that we're waiting for that promise(s) to be resolved. I think it would probably be helpful, but I wanted to see what you think first because it's going to make `promiseAllObject` a little gross because it has to return both the new promise and whether promises were found inside the object or something like that.

Reply:
Can you pass the logger to promiseAllObject and put the logging statements for that inside the promiseAllObject code?

Reply:
We could, but that would probably make `promiseAllObject` aware of and coupled to the context in which it is running. In other words, if inside `promiseAllObject` we had something like:

```
logger.log("Promises found within provided options. Waiting for promises to be resolved before continuing command execution.");
```

It would restrict usage of `promiseAllObject` to this one use case. Making the error more generic might make it too generic. We could do something like this:

```
promiseAllObject(options, logger, "Promises found within provided options. Waiting for promises to be resolved before continuing command execution.");
```

But I think that's worse than just returning an object with both the promise and a boolean indicating whether promises were found (and, probably ideally, where those promises were found in the object structure).

Reply:
That makes sense. I think promiseAllObject(option,logger,"Message") and returning a boolean/nestedPromises are both acceptable options.

Reply:
@jonsnyder I like the approach you've outlined, but one concern I might have is that if someone executed an `event` command twice, and the first one has a promise that gets resolved in, say, 20 milliseconds, and the second `event command doesn't have any promises in its data, the data from the second event will likely get sent to the server before the data from the first event and there won't be any log messages that suggest why that may have happened.